src/coreclr/pal/src/map/virtual.cpp

```diff
@@ -641,7 +641,7 @@ static LPVOID ReserveVirtualMemory(
     }
 #endif  // MMAP_ANON_IGNORES_PROTECTION
 
-#ifdef MADV_DONTDUMP
+#if defined(MADV_DONTDUMP) && !defined(TARGET_BROWSER)
```

Comment:
```suggestion
#if defined(MADV_DONTDUMP) && !defined(TARGET_WASM)
```
Should everything that has to do with memory allocation be WASM instead of BROWSER? The memory allocation seems to be a fundamental wasm concept, not browser specific.